spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/mail/MailSenderAutoConfigurationTests.java

```diff
@@ -149,9 +149,10 @@ public void jndiSessionAvailable() throws NamingException {
 		Session session = configureJndiSession("foo");
 		load(EmptyConfig.class, "spring.mail.jndi-name:foo");
 		Session sessionBean = this.context.getBean(Session.class);
+		JavaMailSenderImpl mailSender = this.context.getBean(JavaMailSenderImpl.class);
 		assertThat(sessionBean).isEqualTo(session);
-		assertThat(this.context.getBean(JavaMailSenderImpl.class).getSession())
-				.isEqualTo(sessionBean);
+		assertThat(mailSender.getSession()).isEqualTo(sessionBean);
+		assertThat(mailSender.getDefaultEncoding()).isEqualTo("UTF-8");
```

Comment:
That test is testing by side effect. Perhaps it should test what the expectations are instead.